app/controllers/api/works_controller.rb

```diff
@@ -97,22 +97,31 @@ def destroy
   end
 
   def index
-    works = Work.all
-    render json: works,
-      each_serializer: WorkSerializer
+    work_count = Work.count
+    works = Work.page(params[:page])
+    render json: {
+      works: ActiveModel::Serializer::CollectionSerializer.new(works, each_serializer: WorkSerializer),
+      work_count: work_count
+    }
   end
 
   def filtered_works
     parsed_query = CGI.parse(params[:search_params])
-    filtered_works = params[:search_params] == "" ?  Work.all : Work.where(parsed_query)
-    render json: filtered_works,
-      each_serializer: WorkSerializer
+    filtered_works = params[:search_params] == "" ?  Work.all.page(params[:page]) : Work.where(parsed_query).page(params[:page])
```

Comment:
don't repeat these sql queries! call `Work.where` just once and use the variable twice :)